src/CoreWCF.Http/tests/ContractBehaviorTests.cs

```diff
@@ -0,0 +1,311 @@
+                string returnStr = clientProxy.StringMethod(HelloStr);
+                Assert.Equal(HelloStr, returnStr);
+            }
+            catch
+            {
+                throw;
+            }
+            finally
+            {
+                if (cf != null && cf.State == System.ServiceModel.CommunicationState.Opened)
+                    cf.Close();
+            }
+        }
+
+        private void Variation_ByHandImplementsOther()
+        {
+            ChannelFactory<IContractBehaviorBasic_ByHand> cf = GetChannelFactory<IContractBehaviorBasic_ByHand>();  
+            try
+            {
+                var theBehavior = new MyMultiFacetedBehaviorAttribute();
+                cf.Endpoint.Contract.ContractBehaviors.Add(theBehavior);
+                cf.Open();
+                string expected = "IContractBehavior:ClientContract.MyMultiFacetedBehaviorAttribute;";
+                Assert.Equal(expected, BehaviorInvokedVerifier.ValidateClientInvokedBehavior(cf.Endpoint));
+                IContractBehaviorBasic_ByHand clientProxy = cf.CreateChannel();
+                string HelloStr = "ByHandImplementsOther";
+                string returnStr = clientProxy.StringMethod(HelloStr);
+                Assert.Equal(HelloStr, returnStr);
+            }
+            catch
+            {
+                throw;
+            }
+            finally
+            {
+                if (cf != null && cf.State == System.ServiceModel.CommunicationState.Opened)
+                    cf.Close();
+            }
+        }
+
+        internal class Startup
+        {
+            public static string _method = "";
+            public void ConfigureServices(IServiceCollection services)
+            {
+                if (_method.Contains("ByHandImplementsOther"))
+                {
+                    services.AddServiceModelServices().AddSingleton<IContractBehavior>(new ServiceContract.MyMultiFacetedBehaviorAttribute());
+                }
+                else if(_method.Contains("ByHand"))
+                {
+                    services.AddServiceModelServices().AddSingleton<IContractBehavior>(new ServiceContract.CustomContractBehaviorAttribute());
+                    
+                }
+                else
+                {
+                    services.AddServiceModelServices();
+                }
+            }
+
+            public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+            {
+                app.UseServiceModel(builder =>
+                {
+                    switch(_method)
+                    {                     
+                        case "ByHand":
+                        case "ByHand_UsingHiddenProperty":
+                        case "ByHandImplementsOther":
+                        case "ByHandImplementsOther_UsingHiddenProperty":
+                            builder.AddService<ContractBehaviorBasic_ByHand_Service>();
```

Comment:
@mconnew , The original .NetFramework test case contains "byhand" test scenarios which added CustomContractBehaviorAttribute by hand via calling method below, do you have any idea on how to do similar config in CoreWCF?

ServiceHost.Description.Endpoints[0].Contract.ContractBehaviors.Add(cb);

Reply:
I've just brought forward and committed a change from some other work to enable this. Check out [this test](https://github.com/CoreWCF/CoreWCF/blob/2e98274e6b50cef235cea4ee5d921eeea663eafe/src/CoreWCF.Http/tests/SimpleTest.cs#L93) to see how to add a delegate which gets access to ServiceHostBase.Description. It's executed just before the equivalent of ServiceHost.Open().

Reply:
Thank you. I'll checkout that and update test.